components/kyma-environment-broker/internal/runtime/handler.go

```diff
@@ -172,6 +175,9 @@ func (h *Handler) setRuntimeAllOperations(instance internal.Instance, dto *pkg.R
 		if len(provOprs) > 1 {
 			h.converter.ApplyUnsuspensionOperations(dto, provOprs[:len(provOprs)-1])
 		}
+		// Set kyma version from the last provisioning operation
+		kymaVersion = lastProvOp.RuntimeVersion.Version
```

Comment:
don't upgrade operations also kyma runtime version? Would it make sense to take the last provisioning or upgrade operation to determine the current kyma version of the SKR?

Reply:
Yes, it's exactly what I'm doing a few lines below this change. If there are upgrade kyma operations and the last upgrade operation happened after the last provision/unsuspension operation, take it's kyma version data instead.

Reply:
Or did I miss something?

Reply:
oh, my bad, I missed that. Ignore me :)